server/paymentProviders/transferwise/index.ts

```diff
@@ -96,31 +101,45 @@ async function quoteExpense(
   }
 
   expense.collective = expense.collective || (await models.Collective.findByPk(expense.CollectiveId));
+  expense.host = expense.host || (await expense.collective.getHostCollective());
   const hasMultiCurrency = expense.currency !== expense.collective.currency;
+  const targetCurrency = payoutMethod.unfilteredData.currency as string;
   const quoteParams = {
     profileId: connectedAccount.data.id,
-    sourceCurrency: expense.collective.currency,
-    targetCurrency: <string>payoutMethod.unfilteredData.currency,
+    sourceCurrency: expense.host.currency,
+    targetCurrency,
     targetAccount,
   };
 
-  // Adapt the fees payer
-  if (expense.feesPayer === 'PAYEE') {
-    // We assume that expense.currency is always collective.currency
-    quoteParams['sourceAmount'] = expense.amount / 100;
-    if (hasMultiCurrency) {
-      throw new Error('Putting the fees on the payee is not yet supported for multi-currency expenses');
-    }
-  } else {
-    // Guarantees the target amount if in the same currency of expense
-    const { rate } = await getTemporaryQuote(connectedAccount, payoutMethod, expense);
-    quoteParams['targetAmount'] = (expense.amount / 100) * rate;
-  }
-
-  // Adapt for multi-currency
   if (hasMultiCurrency) {
+    assert(
+      expense.collective.currency === expense.host.currency,
+      'For multi-currency expenses, the host currency must be the same as the collective currency',
+    );
+    assert(
+      expense.currency === payoutMethod.unfilteredData.currency,
+      'For multi-currency expenses, the payout currency must be the same as the expense currency',
+    );
     quoteParams['targetCurrency'] = expense.currency;
     quoteParams['targetAmount'] = expense.amount / 100;
+  } else if (expense.feesPayer === 'PAYEE') {
+    assert(
+      expense.host.currency === expense.currency,
+      'For expenses coverede by the payee, the host currency must be the same as the expense currency',
+    );
+    quoteParams['sourceAmount'] = expense.amount / 100;
+  } else {
+    let rate = 1;
+    if (targetCurrency !== expense.host.currency) {
+      const [exchangeRate] = await transferwise.getExchangeRates(
+        connectedAccount,
+        expense.host.currency,
+        targetCurrency,
+      );
+      assert(exchangeRate, `No exchange rate found for ${expense.host.currency} -> ${targetCurrency}`);
+      rate = exchangeRate.rate;
+    }
+    quoteParams['targetAmount'] = (expense.amount / 100) * rate;
```

Comment:
Can the rate fluctuate between the `getExchangeRates` and the actual expense quote?

Reply:
I think it can. But since we're just using this to calculate how much we should send in order to cover the fees I think this is fine.